.github/workflows/documentation.yml

```diff
@@ -0,0 +1,48 @@
+name: Documentation
+
+on:
+  push:
+    branches:
+      - master
+      - documentation-refactoring
+
+jobs:
+  generate_references:
+    runs-on: ubuntu-20.04
+    env:
+      GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+    steps:
+      - uses: actions/checkout@v1
+      - name: Generate documentation
+        run: |
+          # Install dependencies
+          sudo apt-get install git python3-setuptools
+          pip3 install wheel
+          pip3 install sphinx-markdown-builder sphinx
+          export PATH=${PATH}:${HOME}/.local/bin
+
+          # Generate docs
+          cd docs
+          make markdown
+
+          # Initialize Git
+          NAME=$(curl https://api.github.com/users/${GITHUB_ACTOR} | jq -r .name)
+          ID=$(curl https://api.github.com/users/${GITHUB_ACTOR} | jq -r .id)
+          git config --global user.name "${NAME}"
+          git config --global user.email "${ID}+${GITHUB_ACTOR}@users.noreply.github.com"
```

Comment:
`${GITHUB_ACTOR}` is a username, not a name, and `${GITHUB_ACTOR}@github.com` is not associated with any GitHub account. But maybe, this isn't important because the commits are hidden in a way.